File: Sorting/Selection Sort/SelectionSort.js
function selectionSort(array) {
    let minIndex = 0;
    for (let i = 0; i < array.length - 1; i++) {
        minIndex = i;
        for (let j = i; j < array.length; j++) {
            if (array[j] < array[minIndex]) minIndex = j;
        }
        if (minIndex != i)
            [array[i], array[minIndex]] = [array[minIndex], array[i]];
    }
    return array;
}
let nums = [7, 3, 2, 5, 6, 10, 9, 8, 1];
selectionSort(nums);

File: Sorting/Selection Sort/SelectionSort.cpp
#include <iostream>

using namespace std;

void selectionSort(int (&array)[], int n)
{
    int minIndex;
    for (int i = 0; i < n - 2; i++)
    {
        minIndex = i;
        for (int j = i; j < n; j++)
        {
            if (array[j] < array[minIndex])
                minIndex = j;
        }
        if (i != minIndex)
            swap(array[i], array[minIndex]);
    }

    // Print values
    for (int i = 0; i < n; i++)
    {
        cout << array[i] << " ";
    }

    return;
}

int main()
{

    int numbers[] = {7, 3, 2, 5, 6, 10, 9, 8, 1};
    int n = sizeof(numbers) / sizeof(numbers[0]);

    selectionSort(numbers, n);

    return 0;
}